pages/example-6-xss/solution-1.tsx

```diff
@@ -0,0 +1,40 @@
+import { BlogPost, getSpecialBlogPosts } from '../../database/blogPosts';
+import { CommonContent } from './common';
+
+type Props =
+  | {
+      error: string;
+    }
+  | {
+      blogPosts: BlogPost[];
+    };
+
+export default function MissingAuthenticationGssp(props: Props) {
+  return (
+    <div>
+      <CommonContent />
+
+      {'error' in props && <div style={{ color: 'red' }}>{props.error}</div>}
+
+      {'blogPosts' in props &&
+        props.blogPosts.map((blogPost) => {
+          return (
+            <div key={`blog-post-${blogPost.id}`}>
+              <h2>{blogPost.title}</h2>
+              <div>Published: {String(blogPost.isPublished)}</div>
+              <div>{blogPost.textContent}</div>
+            </div>
+          );
+        })}
+    </div>
+  );
+}
+
+export async function getServerSideProps() {
+  const blogPosts = await getSpecialBlogPosts(6);
+  return {
+    props: {
+      blogPosts: blogPosts,
```

Comment:
we can probably return `blogPost` as the prop and simplify the page above, since it's just about a single blog post (on all pages)

I'll take this over

Reply:
Done in 24971a9